### PR TITLE
fix: skipper topologySpreadConstraints.labelSelector

### PR DESCRIPTION
**Pull request by szuecs**
we have 2 deployments and component=ingress identifies the data-plane

### CODE REVIEW

**Comment by AlexanderYastrebov** [comment]
:+1:

File: cluster/manifests/skipper/deployment.yaml [hunk 1]
```diff
@@ -79,7 +79,7 @@ spec:
           whenUnsatisfiable: DoNotSchedule
           labelSelector:
             matchLabels:
-              deployment: skipper-ingress
```
Comment (RomanZavodskikh) [inline cluster/manifests/skipper/deployment.yaml:82]: Should we add application label as well?

**Comment by AlexanderYastrebov** [comment]
Does it matter though? The second deployment is a single canary instance.

File: cluster/manifests/skipper/deployment.yaml (hunk 1 above)
Comment (szuecs) [reply cluster/manifests/skipper/deployment.yaml:82]: good idea!

**Comment by szuecs** [comment]
> Does it matter though? The second deployment is a single canary instance.

The problem is that we set `maxSkew 1`, but with canary deployment we really have 1 not counted so we can assume always  `maxSkew 2` and we can fix it by changing the selector.

File: cluster/manifests/skipper/deployment.yaml (hunk 1 above)
Comment (szuecs) [reply cluster/manifests/skipper/deployment.yaml:82]: fixed

**Comment by RomanZavodskikh** [comment]
:+1:

**Comment by AlexanderYastrebov** [comment]
:+1:

**Pull request merged**